From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Molean <cnmolean@gmail.com>
Date: Sun, 3 Jul 2022 01:08:56 +0800
Subject: [PATCH] region monitor


diff --git a/src/main/java/com/molean/region/monitor/RegionMonitor.java b/src/main/java/com/molean/region/monitor/RegionMonitor.java
new file mode 100644
index 0000000000000000000000000000000000000000..d44322189f5a62f10a4217c775d77e012753ef12
--- /dev/null
+++ b/src/main/java/com/molean/region/monitor/RegionMonitor.java
@@ -0,0 +1,63 @@
+package com.molean.region.monitor;
+
+import java.util.Deque;
+import java.util.LinkedList;
+
+public class RegionMonitor {
+    private RegionMonitorManager regionMonitorManager;
+    private final int x;
+    private final int z;
+
+    private Deque<RegionTickData> queue = new LinkedList<>();
+
+    public static class RegionTickData {
+        long tick;
+        long nanoTime;
+        long lastStart;
+
+        public long getTick() {
+            return tick;
+        }
+
+        public long getNanoTime() {
+            return nanoTime;
+        }
+    }
+
+    public RegionMonitor(RegionMonitorManager manager, int x, int z) {
+        this.x = x;
+        this.z = z;
+    }
+
+    public void startTick(long currentTick) {
+        RegionTickData regionTickData;
+        if (!queue.isEmpty() && queue.getLast().tick == currentTick) {
+            regionTickData = queue.getLast();
+        } else {
+            regionTickData = new RegionTickData();
+            regionTickData.tick = currentTick;
+            queue.addLast(regionTickData);
+            while (!queue.isEmpty() && currentTick - queue.getFirst().tick > 1200) {
+                queue.removeFirst();
+            }
+        }
+        regionTickData.lastStart = System.nanoTime();
+    }
+
+    public void stopTick() {
+        RegionTickData last = queue.getLast();
+        last.nanoTime += System.nanoTime() - last.lastStart;
+    }
+
+    public int getX() {
+        return x;
+    }
+
+    public int getZ() {
+        return z;
+    }
+
+    public Deque<RegionTickData> getQueue() {
+        return queue;
+    }
+}
diff --git a/src/main/java/com/molean/region/monitor/RegionMonitorManager.java b/src/main/java/com/molean/region/monitor/RegionMonitorManager.java
new file mode 100644
index 0000000000000000000000000000000000000000..40c755beb8ab9e7e0bd5f723623f364a1fd45f9e
--- /dev/null
+++ b/src/main/java/com/molean/region/monitor/RegionMonitorManager.java
@@ -0,0 +1,24 @@
+package com.molean.region.monitor;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public enum RegionMonitorManager {
+    INSTANCE;
+
+    private long tick;
+
+    private final Map<RegionPos, RegionMonitor> regionMonitorMap = new HashMap<>();
+
+    public Map<RegionPos, RegionMonitor> getRegionMonitorMap() {
+        return regionMonitorMap;
+    }
+
+    public RegionMonitor get(int x, int z) {
+        RegionPos regionPos = new RegionPos(x, z);
+        if (!regionMonitorMap.containsKey(regionPos)) {
+            regionMonitorMap.put(regionPos, new RegionMonitor(this, x, z));
+        }
+        return regionMonitorMap.get(regionPos);
+    }
+}
diff --git a/src/main/java/com/molean/region/monitor/RegionPos.java b/src/main/java/com/molean/region/monitor/RegionPos.java
new file mode 100644
index 0000000000000000000000000000000000000000..e654fedde3b641d08df0cd603a7e3b1ada721ddd
--- /dev/null
+++ b/src/main/java/com/molean/region/monitor/RegionPos.java
@@ -0,0 +1,53 @@
+package com.molean.region.monitor;
+
+public class RegionPos {
+    private int x;
+    private int z;
+
+    public RegionPos(int x, int z) {
+        this.x = x;
+        this.z = z;
+    }
+
+    public int getX() {
+        return x;
+    }
+
+    public int getZ() {
+        return z;
+    }
+
+    public static RegionPos fromChunk(int x, int z) {
+        return new RegionPos(x >> 5, z >> 5);
+    }
+
+    public static RegionPos fromBlock(int x, int z) {
+        return new RegionPos(x >> 9, z >> 9);
+    }
+
+    @Override
+    public String toString() {
+        return "RegionPos{" +
+            "x=" + x +
+            ", z=" + z +
+            '}';
+    }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) return true;
+        if (object == null || getClass() != object.getClass()) return false;
+
+        RegionPos regionPos = (RegionPos) object;
+
+        if (x != regionPos.x) return false;
+        return z == regionPos.z;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = x;
+        result = 31 * result + z;
+        return result;
+    }
+}
