From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Molean <cnmolean@gmail.com>
Date: Sun, 10 Jul 2022 20:19:50 +0800
Subject: [PATCH] custom structure register


diff --git a/src/main/java/com/molean/structures/CustomStructureRegister.java b/src/main/java/com/molean/structures/CustomStructureRegister.java
new file mode 100644
index 0000000000000000000000000000000000000000..bd0bda238a88a1c0c92f42536e0d907e10516330
--- /dev/null
+++ b/src/main/java/com/molean/structures/CustomStructureRegister.java
@@ -0,0 +1,29 @@
+package com.molean.structures;
+
+import com.molean.regionmonitor.RegionPos;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public enum CustomStructureRegister {
+    INSTANCE,
+    ;
+
+    CustomStructureRegister() {
+    }
+
+    Map<RegionPos, String> structureMap = new HashMap<>();
+
+
+    public void register(int regionX, int reginZ, String structureType) {
+        structureMap.put(new RegionPos(regionX, reginZ), structureType);
+    }
+
+    public String getStructure(int x, int z) {
+        return structureMap.get(new RegionPos(x, z));
+    }
+
+    public Map<RegionPos, String> getStructureMap() {
+        return structureMap;
+    }
+}
