From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Molean <cnmolean@gmail.com>
Date: Fri, 9 Dec 2022 01:04:06 +0800
Subject: [PATCH] fix crash


diff --git a/src/main/java/com/molean/structures/CustomStructureRegister.java b/src/main/java/com/molean/structures/CustomStructureRegister.java
index af851caaed1f719d4ecd14eec47e4c2e03832a4e..75018dfe58bd72038fcf3d9d5656eba2f0b78ac6 100644
--- a/src/main/java/com/molean/structures/CustomStructureRegister.java
+++ b/src/main/java/com/molean/structures/CustomStructureRegister.java
@@ -19,7 +19,7 @@ public enum CustomStructureRegister {
 
     private final Map<LevelRegionPos, String> levelStructureMap = new HashMap<>();
 
-    private BiFunction<String, Pair<Integer, Integer>, String> predictor;
+    private BiFunction<String, Pair<Integer, Integer>, String> predictor = (s, integerIntegerPair) -> null;
 
     public void setPredictor(BiFunction<String, Pair<Integer, Integer>, String> predictor) {
         this.predictor = predictor;
