From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Molean <cnmolean@gmail.com>
Date: Sun, 3 Jul 2022 01:08:56 +0800
Subject: [PATCH] region monitor


diff --git a/src/main/java/net/minecraft/server/level/ServerLevel.java b/src/main/java/net/minecraft/server/level/ServerLevel.java
index e13fcd15ab00ab4aee7fb3774db6984743680259..95a32fb30987cfb7b56a7eb254cf62ecfb0476cb 100644
--- a/src/main/java/net/minecraft/server/level/ServerLevel.java
+++ b/src/main/java/net/minecraft/server/level/ServerLevel.java
@@ -6,6 +6,8 @@ import com.google.common.collect.Lists;
 import com.mojang.datafixers.DataFixer;
 import com.mojang.datafixers.util.Pair;
 import com.mojang.logging.LogUtils;
+import com.molean.regionmonitor.RegionMonitor;
+import com.molean.regionmonitor.RegionMonitorManager;
 import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
 import it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap;
 import it.unimi.dsi.fastutil.longs.LongSet;
@@ -670,6 +672,8 @@ public class ServerLevel extends Level implements WorldGenLevel {
             org.spigotmc.ActivationRange.activateEntities(this); // Spigot
             timings.entityTick.startTiming(); // Spigot
             this.entityTickList.forEach((entity) -> {
+                RegionMonitor regionMonitor = RegionMonitorManager.INSTANCE.get(entity.getBlockX() >> 9, entity.getBlockZ() >> 9); //Islet
+                regionMonitor.startTick(entity.getLevel().getCraftServer().getCurrentTick()); //Islet
                 if (!entity.isRemoved()) {
                     if (false && this.shouldDiscardEntity(entity)) { // CraftBukkit - We prevent spawning in general, so this butchering is not needed
                         entity.discard();
@@ -694,6 +698,7 @@ public class ServerLevel extends Level implements WorldGenLevel {
                         }
                     }
                 }
+                regionMonitor.stopTick(); // Islet
             });
             timings.entityTick.stopTiming(); // Spigot
             timings.tickEntities.stopTiming(); // Spigot
diff --git a/src/main/java/net/minecraft/world/level/Level.java b/src/main/java/net/minecraft/world/level/Level.java
index d59dea221ba0f1b9c14f403d3c6ea61b2c454316..31816771834345761630c5028cf0bbf7cd74fec8 100644
--- a/src/main/java/net/minecraft/world/level/Level.java
+++ b/src/main/java/net/minecraft/world/level/Level.java
@@ -15,6 +15,8 @@ import java.util.function.Consumer;
 import java.util.function.Predicate;
 import java.util.function.Supplier;
 import javax.annotation.Nullable;
+import com.molean.regionmonitor.RegionMonitor;
+import com.molean.regionmonitor.RegionMonitorManager;
 import net.minecraft.CrashReport;
 import net.minecraft.CrashReportCategory;
 import net.minecraft.ReportedException;
@@ -870,6 +872,8 @@ public abstract class Level implements LevelAccessor, AutoCloseable {
             }
             // Spigot end
 
+            RegionMonitor regionMonitor = RegionMonitorManager.INSTANCE.get(tickingblockentity.getPos().getX() >> 9, tickingblockentity.getPos().getZ() >> 9);
+            regionMonitor.startTick(getCraftServer().getCurrentTick());
             if (tickingblockentity.isRemoved()) {
                 // Spigot start
                 tilesThisCycle--;
@@ -883,6 +887,7 @@ public abstract class Level implements LevelAccessor, AutoCloseable {
                 }
                 // Paper end - execute chunk tasks during tick
             }
+            regionMonitor.stopTick();
         }
         this.blockEntityTickers.removeAll(toRemove);
 
diff --git a/src/main/java/net/minecraft/world/level/NaturalSpawner.java b/src/main/java/net/minecraft/world/level/NaturalSpawner.java
index e31a2eea9a62ab2c0bed1a97dab6bae231b8cd8b..6d637db2d595593fb60b2c36ff737019a25e05d6 100644
--- a/src/main/java/net/minecraft/world/level/NaturalSpawner.java
+++ b/src/main/java/net/minecraft/world/level/NaturalSpawner.java
@@ -1,6 +1,9 @@
 package net.minecraft.world.level;
 
 import com.mojang.logging.LogUtils;
+import com.molean.regionmonitor.RegionMonitor;
+import com.molean.regionmonitor.RegionMonitorManager;
+import com.molean.regionmonitor.RegionPos;
 import it.unimi.dsi.fastutil.objects.Object2IntMap;
 import it.unimi.dsi.fastutil.objects.Object2IntMaps;
 import it.unimi.dsi.fastutil.objects.Object2IntOpenHashMap;
@@ -133,6 +136,8 @@ public final class NaturalSpawner {
     public static void spawnForChunk(ServerLevel world, LevelChunk chunk, NaturalSpawner.SpawnState info, boolean spawnAnimals, boolean spawnMonsters, boolean rareSpawn) {
         world.getProfiler().push("spawner");
         world.timings.mobSpawn.startTiming(); // Spigot
+        RegionMonitor regionMonitor = RegionMonitorManager.INSTANCE.get(chunk.getPos().x >> 5, chunk.getPos().x >> 5); // Islet
+        regionMonitor.startTick(world.getCraftServer().getCurrentTick()); // Islet
         MobCategory[] aenumcreaturetype = NaturalSpawner.SPAWNING_CATEGORIES;
         int i = aenumcreaturetype.length;
 
@@ -187,6 +192,7 @@ public final class NaturalSpawner {
             }
         }
 
+        regionMonitor.stopTick(); //Islet
         world.timings.mobSpawn.stopTiming(); // Spigot
         world.getProfiler().pop();
     }
