From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Molean <cnmolean@gmail.com>
Date: Sun, 14 Aug 2022 18:18:51 +0800
Subject: [PATCH] disable natural spawn for region


diff --git a/src/main/java/net/minecraft/server/level/ServerLevel.java b/src/main/java/net/minecraft/server/level/ServerLevel.java
index 58d7345c98dfe45fd23fcde8f6aab504120a3ae5..55acac9d4eec4957986b3c3b261c05785fd00865 100644
--- a/src/main/java/net/minecraft/server/level/ServerLevel.java
+++ b/src/main/java/net/minecraft/server/level/ServerLevel.java
@@ -6,6 +6,7 @@ import com.google.common.collect.Lists;
 import com.mojang.datafixers.DataFixer;
 import com.mojang.datafixers.util.Pair;
 import com.mojang.logging.LogUtils;
+import com.molean.region.modifier.NaturalSpawningDisableManager;
 import com.molean.region.monitor.RegionMonitor;
 import com.molean.region.monitor.RegionMonitorManager;
 import com.molean.region.monitor.RegionPos;
diff --git a/src/main/java/net/minecraft/world/level/NaturalSpawner.java b/src/main/java/net/minecraft/world/level/NaturalSpawner.java
index 55f86bc892b47764d6487f932c520cb849158940..29aca952dba5775f4b2a3f4220f2c7370866c8a7 100644
--- a/src/main/java/net/minecraft/world/level/NaturalSpawner.java
+++ b/src/main/java/net/minecraft/world/level/NaturalSpawner.java
@@ -1,6 +1,7 @@
 package net.minecraft.world.level;
 
 import com.mojang.logging.LogUtils;
+import com.molean.region.modifier.NaturalSpawningDisableManager;
 import com.molean.region.monitor.RegionMonitor;
 import com.molean.region.monitor.RegionMonitorManager;
 import com.molean.region.monitor.RegionPos;
@@ -135,6 +136,9 @@ public final class NaturalSpawner {
     }
 
     public static void spawnForChunk(ServerLevel world, LevelChunk chunk, NaturalSpawner.SpawnState info, boolean spawnAnimals, boolean spawnMonsters, boolean rareSpawn) {
+        if (NaturalSpawningDisableManager.INSTANCE.isRegionNaturalSpawningDisabled(new RegionPos(chunk.getPos().getRegionX(), chunk.getPos().getRegionZ()))) {
+            return;
+        }
         world.getProfiler().push("spawner");
         world.timings.mobSpawn.startTiming(); // Spigot
         MobCategory[] aenumcreaturetype = NaturalSpawner.SPAWNING_CATEGORIES;
